sources/packages/backend/apps/api/src/institution-integration/ier-integration/ier12-file-detail.ts

```diff
@@ -0,0 +1,140 @@
+    const record = new StringBuilder();
+    record.appendWithStartFiller(this.assessmentId, 10, NUMBER_FILLER);
+    record.append(this.applicationNumber, 10);
+    record.append(this.sin, 9);
+    record.appendWithEndFiller(this.studentLastName, 25, SPACE_FILLER);
+    record.appendWithEndFiller(this.studentGivenName ?? "", 15, SPACE_FILLER);
+    record.appendDate(this.birthDate, DATE_FORMAT);
+    record.appendWithEndFiller(this.programName, 25, SPACE_FILLER);
+    record.appendWithEndFiller(this.programDescription, 50, SPACE_FILLER);
+    record.appendWithEndFiller(this.credentialType, 25, SPACE_FILLER);
+    record.append(this.cipCode.toString(), 6);
+    record.append(this.nocCode, 4);
+    record.appendWithEndFiller(this.sabcCode ?? "", 4, SPACE_FILLER);
+    record.appendWithEndFiller(
+      this.institutionProgramCode ?? "",
+      25,
+      SPACE_FILLER,
+    );
+    record.append(this.programLength.toString(), 1);
+    record.appendDate(this.studyStartDate, DATE_FORMAT);
+    record.appendDate(this.studyEndDate, DATE_FORMAT);
+    record.append(this.tuitionFees.toString(), 8);
+    record.append(this.programRelatedCosts.toString(), 8);
+    record.append(this.mandatoryFees.toString(), 8);
+    record.append(this.exceptionExpenses.toString(), 8);
+    record.append(this.totalFundedWeeks.toString(), 2);
+    record.repeatAppend(SPACE_FILLER, 3);
+    record.append("F", 1);
+    record.appendWithStartFiller(this.populateGrants(CSLF), 8, NUMBER_FILLER);
+    record.appendWithStartFiller(this.populateGrants(CSGP), 8, NUMBER_FILLER);
+    record.appendWithStartFiller(this.populateGrants(CSGD), 8, NUMBER_FILLER);
+    record.appendWithStartFiller(this.populateGrants(CSGF), 8, NUMBER_FILLER);
+    record.appendWithStartFiller(this.populateGrants(CSGT), 8, NUMBER_FILLER);
+    record.appendWithStartFiller(this.populateGrants(BCSL), 8, NUMBER_FILLER);
+    record.appendWithStartFiller(this.populateGrants(BCAG), 8, NUMBER_FILLER);
+    record.appendWithStartFiller(this.populateGrants(BGPD), 8, NUMBER_FILLER);
+    record.appendWithStartFiller(this.populateGrants(SBSD), 8, NUMBER_FILLER);
+    record.appendWithStartFiller(this.populateGrants(BCSG), 8, NUMBER_FILLER);
+    return record.toString();
+  }
+
+  /**
+   * Populate the grants for the given grant code.
+   * @param valueCode
+   * @returns Rounded value amount for the grant.
+   */
+  private populateGrants(valueCode: string) {
+    return round(
+      parseInt(
+        this.totalGrants.find(
+          (disbursementValue) => (disbursementValue.valueCode = valueCode),
+        ).valueAmount,
+      ),
+    );
+  }
+
+  /**
+   * Creates an accumulated disbursement values of the multiple disbursement
+   * schedules for the particular assessment.
+   * @returns totalGrants from the disbursementSchedules.
+   */
+  private getTotalGrantsFromDisbursementSchedule(): DisbursementValue[] {
+    const accumulatedDisbursementSchedule = Object.values(
```

Comment:
I am a bit confused. what exactly are we doing here

Reply:
refactored with flatmap, now it should be clear.